core/trino-main/src/main/java/io/trino/sql/gen/CursorProcessorCompiler.java

```diff
@@ -137,14 +143,27 @@ private Class<? extends CursorProcessor> compile(Optional<RowExpression> filter,
 
     private Class<? extends CursorProcessor> compileProcessor(RowExpression filter, List<RowExpression> projections)
     {
+        CallSiteBinder callSiteBinder = new CallSiteBinder();
+
+        // Projections are split into multiple classes to prevent the process method from exceeding the JVM method size limit
+        // and to avoid hitting the constant pool size limit in the generated CursorProcessor.
+        List<CursorProcessorChunkClass> chunkClasses = new ArrayList<>();
+        for (int projectionIndex = 0; projectionIndex < projections.size(); projectionIndex += MAX_PROJECTIONS_PER_CHUNK) {
```

Comment:
Seems you should be able to make it nicer using `Lists.partition` from Guava

Reply:
Good idea. Suggestion applied. However, I didn't chain calls because I need to track the projection index (required to generate method names - `project_1`, `project_2`, ...